nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1995,13 +1995,17 @@ public static DataType getDataTypeFromSQLTypeValue(final int sqlType) {
             case Types.NVARCHAR:
             case Types.OTHER:
             case Types.SQLXML:
+            case Types.CLOB:
                 return RecordFieldType.STRING.getDataType();
             case Types.TIME:
                 return RecordFieldType.TIME.getDataType();
             case Types.TIMESTAMP:
                 return RecordFieldType.TIMESTAMP.getDataType();
             case Types.ARRAY:
                 return RecordFieldType.ARRAY.getDataType();
+            case Types.BINARY:
+            case Types.BLOB:
+                return RecordFieldType.ARRAY.getArrayDataType(RecordFieldType.BYTE.getDataType());
```

Comment:
It would be helpful to have a unit test for the changes to indicate the intended behavior.